File: Chapter3_VB/Student.vb
Option Strict On

Public Class Student
    Private _firstName As String = "Erick"

    Public Property LastName As String = "Kurniawan"

    Property FirstName() As String
        Get
            Return _firstName
        End Get
        Set(ByVal value As String)
            _firstName = value
        End Set
    End Property

    Public Function getFullName() As String
        Return _firstName & " " & _lastName
    End Function

    Public Function getFullName(firstName As String, lastName As String) As String
        Return firstName & " " & lastName
    End Function

    Public Function getFullName(firstName As String, lastName As String, address As String) As String
        Return firstName & " " & lastName & " " & address
    End Function
End Class

File: Chapter3/Student.cs
namespace SampleCSharp
{
    public class Student
    {
        public string LastName { get; set; } = string.Empty;

        private string firstName = string.Empty;
        public string FirstName
        {
            get
            {
                return firstName;
            }
            set
            {
                if (value.Length > 0)
                    firstName = value;
                else
                    firstName = string.Empty;
            }
        }

        public string getFirstName()
        {
            return firstName;
        }

        public void setFirstName(string value)
        {
            if (value.Length > 0)
                firstName = value;
            else
                firstName = string.Empty;
        }

        public string getFullName()
        {
            return $"{firstName}";
        }

        public string getFullName(string firstName, string lastName)
        {
            return $"{firstName} {lastName}";
        }

        public string getFullName(string firstName, string lastName, string address)
        {
            return $"{firstName} {lastName} {address}";
        }
    }
}
